torchvision/prototype/transforms/_augment.py

```diff
@@ -0,0 +1,149 @@
+import math
+import numbers
+import warnings
+from typing import Any, Dict, Tuple
+
+import torch
+from torchvision.prototype.transforms import Transform
+from torchvision.prototype.utils._internal import query_recursively
+
+from . import functional as F
+
+
+class RandomErasing(Transform):
+    _DISPATCHER = F.erase
+
+    def __init__(
+        self,
+        p: float = 0.5,
+        scale: Tuple[float, float] = (0.02, 0.33),
+        ratio: Tuple[float, float] = (0.3, 3.3),
+        value: float = 0,
+    ):
+        super().__init__()
+        if not isinstance(value, (numbers.Number, str, tuple, list)):
+            raise TypeError("Argument value should be either a number or str or a sequence")
+        if isinstance(value, str) and value != "random":
+            raise ValueError("If value is str, it should be 'random'")
+        if not isinstance(scale, (tuple, list)):
+            raise TypeError("Scale should be a sequence")
+        if not isinstance(ratio, (tuple, list)):
+            raise TypeError("Ratio should be a sequence")
+        if (scale[0] > scale[1]) or (ratio[0] > ratio[1]):
+            warnings.warn("Scale and ratio should be of kind (min, max)")
+        if scale[0] < 0 or scale[1] > 1:
+            raise ValueError("Scale should be between 0 and 1")
+        if p < 0 or p > 1:
+            raise ValueError("Random erasing probability should be between 0 and 1")
+        # TODO: deprecate p in favor of wrapping the transform in a RandomApply
+        self.p = p
+        self.scale = scale
+        self.ratio = ratio
+        self.value = value
+
+    def _get_params(self, sample: Any) -> Dict[str, Any]:
+        image: Any = next(
+            query_recursively(
+                lambda input: input if input in self._DISPATCHER else None, sample  # type: ignore[no-any-return]
```

Comment:
Nit: Though the naming of the variable to `image` was helpful to understand that indeed you were fetching an image, this snippet was not immediately obvious to me how it worked. It works because the `F.erase` is only defined at the moment only for `torch.Tensor` and `features.Image`. How do you feel above requesting the first available image more explicitly? Also you can consider turning this into a utility method given it's being used a few times in the code-base.